src/redux/actions/postingActions.js

```diff
@@ -3,36 +3,151 @@ import {
   DELETE_POSTING,
   LOAD_ITEM_DETAIL,
   UPDATE_ITEM_DETAIL,
+  LOAD_POSTINGS,
+  ERROR_SNACKBAR,
 } from "../constants/actionTypes";
+import { setLoading } from "./loadingActions";
+import axios from "axios";
 
-export const addPosting = (posting, currentUser) => {
+const BASE_URL = `${process.env.REACT_APP_BASE_URL}`;
+
+const addPostingSuccess = (posting) => {
   return {
     type: ADD_POSTING,
     posting: posting,
-    currentUser: currentUser,
   };
 };
 
-export const loadItemDetail = (itemId, postings) => {
+export const displayError = (msg) => {
   return {
-    type: LOAD_ITEM_DETAIL,
-    itemId: itemId,
-    // TODO: postings unnecessary for when BE integrated
+    type: ERROR_SNACKBAR,
+    msg: msg,
+  };
+};
+
+const loadAllPostingsSuccess = (postings) => {
+  return {
+    type: LOAD_POSTINGS,
     postings: postings,
   };
 };
 
-export const updateItemDetail = (itemId, details) => {
+const loadItemDetailSuccess = (item) => {
+  return {
+    type: LOAD_ITEM_DETAIL,
+    item: item,
+  };
+};
+
+const updateItemDetailSuccess = (itemId, details) => {
   return {
     type: UPDATE_ITEM_DETAIL,
     itemId: itemId,
     details: details,
   };
 };
 
-export const deletePosting = (itemId) => {
+const deletePostingSuccess = (itemId) => {
   return {
     type: DELETE_POSTING,
     itemId: itemId,
   };
 };
+
+export const addPosting = (posting, currentUser) => {
+  return async (dispatch) => {
+    try {
+      dispatch(setLoading());
+      let postingRequest = {
+        ...posting,
+        ownerId: currentUser._id,
+        // ownerUsername: currentUser.userName,
+        date: new Date(),
+        location: currentUser.location
+          ? currentUser.location
+          : "The Darkest Timeline",
+      };
+      let postingResponse = await axios.post(
+        `${BASE_URL}/postings`,
+        postingRequest
+      );
+      dispatch(addPostingSuccess(postingResponse.data));
+    } catch (error) {
+      dispatch(displayError("Something went wrong. Posting was not created"));
+    }
+  };
+};
+
+export const loadAllPostings = () => {
+  return async (dispatch) => {
+    try {
+      dispatch(setLoading());
+      let postingResponse = await axios.get(`${BASE_URL}/postings`);
+      dispatch(loadAllPostingsSuccess(postingResponse.data));
+    } catch (error) {
+      dispatch(displayError("Something went wrong. Posting was not deleted"));
+    }
+  };
+};
+
+export const loadPostingsByQuery = (query) => {
+  return async (dispatch) => {
+    try {
+      dispatch(setLoading());
+      let url = query
+        ? `${BASE_URL}/postings/search/${query}`
+        : `${BASE_URL}/postings`;
+      let postingResponse = await axios.get(url);
+      dispatch(loadAllPostingsSuccess(postingResponse.data));
+    } catch (error) {
+      dispatch(displayError("Something went wrong. Posting was not deleted"));
```

Comment:
Shouldn't be a message about deleting. Should be something like "Something went wrong. Please try reloading the page" (or something along those lines I think)